jvm-agent/src/main/org/jetbrains/kotlinx/lincheck/transformation/OwnerNameAnalyzer.kt

```diff
@@ -0,0 +1,814 @@
+        super.visitInsn(opcode)
+        execute(opcode, 0, null)
+        if ((opcode >= Opcodes.IRETURN && opcode <= Opcodes.RETURN) || opcode == Opcodes.ATHROW) {
+            this.locals = null
+            this.stack = null
+        }
+    }
+
+    override fun visitIntInsn(opcode: Int, operand: Int) {
+        super.visitIntInsn(opcode, operand)
+        execute(opcode, operand, null)
+    }
+
+    override fun visitVarInsn(opcode: Int, varIndex: Int) {
+        super.visitVarInsn(opcode, varIndex)
+        val stackSlotSize = if (opcode == Opcodes.RET) 1 else getLocalVarAccessOpcodeType(opcode).size
+        maxLocals = max(maxLocals, varIndex + stackSlotSize)
+        execute(opcode, varIndex, null)
+    }
+
+    override fun visitTypeInsn(opcode: Int, type: String) {
+        super.visitTypeInsn(opcode, type)
+        execute(opcode, 0, type)
+    }
+
+    override fun visitFieldInsn(opcode: Int, owner: String?, name: String?, descriptor: String) {
+        super.visitFieldInsn(opcode, owner, name, descriptor)
+        execute(opcode, 0, className = owner, fieldName = name, descriptor = descriptor)
+    }
+
+    override fun visitMethodInsn(
+        opcodeAndSource: Int,
+        owner: String?,
+        name: String,
+        descriptor: String,
+        isInterface: Boolean
+    ) {
+        if (api < Opcodes.ASM5 && (opcodeAndSource and Opcodes.SOURCE_DEPRECATED) == 0) {
+            // Redirect the call to the deprecated version of this method.
+            super.visitMethodInsn(opcodeAndSource, owner, name, descriptor, isInterface)
+            return
+        }
+        super.visitMethodInsn(opcodeAndSource, owner, name, descriptor, isInterface)
+        val opcode = opcodeAndSource and Opcodes.SOURCE_MASK.inv()
+
+        if (!isInitialized) {
+            return
+        }
+
+        val returnType = Type.getReturnType(descriptor)
+        pop(descriptor)
+        if (opcode != Opcodes.INVOKESTATIC) {
+            pop()
+        }
+        push(returnType.size)
+    }
+
+    override fun visitInvokeDynamicInsn(
+        name: String?,
+        descriptor: String,
+        bootstrapMethodHandle: Handle?,
+        vararg bootstrapMethodArguments: Any?
+    ) {
+        super.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, *bootstrapMethodArguments)
+        if (!isInitialized) {
+            return
+        }
+
+        val returnType = Type.getReturnType(descriptor)
+        pop(descriptor)
+        push(returnType.size)
+    }
+
+    override fun visitJumpInsn(opcode: Int, label: Label?) {
+        super.visitJumpInsn(opcode, label)
+        execute(opcode, 0, null)
+        if (opcode == Opcodes.GOTO) {
+            this.locals = null
+            this.stack = null
+        }
+    }
+
+    override fun visitLdcInsn(value: Any?) {
+        super.visitLdcInsn(value)
+        if (!isInitialized) {
+            return
+        }
+        // TODO: need to investigate how to lookup constant name (if available)
```

Comment:
Are these tickets for these TODOs?

Reply:
https://youtrack.jetbrains.com/issue/DR-310/Support-constants-in-OwnerNameAnalyzerAdapter